tictactoe.py

```diff
@@ -0,0 +1,73 @@
+class Tic_Tac_Toe:
+    def __init__ (self):
+        self.board = ['1','2','3','4','5','6','7','8','9']
+        self.options = ['1','2','3','4','5','6','7','8','9']
+        self.num_of_turns = 0 
+
+
+    def display_board(self):
+        print(self.board[0] + ' | ' + self.board[1] + ' | ' + self.board[2])
+        print("------------")
+        print(self.board[3] + ' | ' + self.board[4] + ' | ' + self.board[5])
+        print("------------")
+        print(self.board[6] + ' | ' + self.board[7] + ' | ' + self.board[8])
+
+    def play_game(self): #There must be two players
+        self.display_board()
+        while (self.winner() is not True):
```

Comment:
I like the general game loop but I don't like how you litter the rest of your logic code with things that your core game loop already does.

Here's my understanding of the core game loop
```
1. render_board
2. ask_for_input
3. check_game_end (includes tie condition) (possibly in turns)
```
I don't like how your `turns` method is taking on the responsibility of the core main loop by rendering the board after some condition, that shouldn't be needed.
In my mind `turns` should only be responsible for updating game state
some other method `render` should be responsible for rendering game state / instructions. You should think about making your `display_board` that `render` method.
What this means is that your player turn needs to move up in game state, owned by TicTacToe `self.curr_player = 1 or -1 or however you want to keep track of player where self.turns can set self.curr_player *= -1 after each successful turn`

The core of what I'm trying to suggest for this and any future project is think of keeping your concerns of ownership separate. In this game loop, we want to think of keeping rendering and game logic separate.